cli/args/flags.rs

```diff
@@ -1109,6 +1110,18 @@ impl Flags {
     })
   }
 
+  pub fn cache_setting(&self) -> CacheSetting {
```

Comment:
needed to break the factory dependency cycle between cli_options and the registry api provider